test/persistence/mzworkflows.py

```diff
@@ -152,3 +156,66 @@ def workflow_disable_user_indexes(w: Workflow):
         destroy_volumes=True,
     )
     w.remove_volumes(volumes=["mzdata"])
+
+def kafka_sources_load_test_step(w: Workflow, current_message_count: int):
+    w.kill_services(services=["materialized"], signal="SIGKILL")
+    w.run_service(
+        service="kgen",
+        command=" ".join(
+            [
+                "--num-records=1000000",
+                "--keys=random",
+                "--key-min=1",
+                "--key-max=10000000000000000000",
+                "--values=bytes",
+                "--max-message-size=501",
+                "--min-message-size=500",
+                "--topic=testdrive-load-test-1",
+            ]
+        )
+    )
+
+    w.start_services(services=["materialized"])
+    w.wait_for_mz(service="materialized")
+    curr_time = time.time()
+    message_count = current_message_count + 1000000
+    print(f"started mz at {curr_time}")
+
+    host_port = w.composition.find_host_ports("materialized")[0]
+    conn = psycopg2.connect(f"host=localhost port={host_port} user=materialize")
+    conn.autocommit = True
+
+    cur = conn.cursor()
+
+    while True:
+        try:
+            cur.execute("SELECT * FROM count")
+            n = cur.fetchone()[0]
+            cur_time = time.time()
+            print(f"found {n} records at {cur_time}")
+            if n == message_count:
+                break
+        except (
+            psycopg2.errors.SqlStatementNotYetComplete,
+            psycopg2.errors.InternalError,
+        ):
+            pass
+        time.sleep(1)
```

Comment:
We can definitely poll a materialized view more than once per second to get better granularity on the benchmark results (I hope?) i wonder what testdrive does

Reply:
testdrive retries the request with exponential backoff with a factor of 1.5 starting with a default initial backoff of 50 ms. Its a different problem domain because testdrive isn't trying to measure the the time elapsed for some event.

I chose 1 second because of a few reasons:

1. When you have ~10 - 20 million records which I feel like is the minimum size for a benchmark like this the restarts take on the order of minutes, so I felt the measurement error on the elapsed times induced by the polling frequency was minimal

2. 1 second is the default interval at which we mint new timestamps, so if we polled say every 10 ms, 99 out of 100 selects guaranteed to return the same count so we wouldn't actually get more granular data about how fast we're reading from the topic, we can only reduce the measurement error on the timing.

3. It's what the existing avro ingest benchmark does (not really a good reason).

Happy to change to a smaller interval - since we're reading from a predefined materialized view the selects should be fast.

Reply:
Nah that's sufficiently compelling, let's include this as a comment in the code plz. (IMO more 2 than 1 or 3, especially since it shouldn't take us minutes to deal with O(10m) things and eventually it won't.)

Reply:
done